File: articles/calculus-for-systems-modeling/articles/energy-balance-models/fortran/energy_balance_models.f90
program energy_balance_models
  implicit none
  real(8) :: forcing, feedback, heat_capacity, equilibrium, tau
  forcing = 3.7d0
  feedback = 1.2d0
  heat_capacity = 10.0d0
  equilibrium = forcing / feedback
  tau = heat_capacity / feedback
  print '(A)', 'scenario equilibrium_temperature adjustment_time'
  print '(A,1X,F12.6,1X,F12.6)', 'baseline_one_layer', equilibrium, tau
end program

File: articles/calculus-for-systems-modeling/articles/energy-balance-models/cpp/energy_balance_models.cpp
#include <iostream>
double equilibrium_temperature(double forcing, double feedback){ return forcing / feedback; }
double adjustment_time(double heat_capacity, double feedback){ return heat_capacity / feedback; }
double absorbed_solar(double solar_constant, double albedo){ return solar_constant * (1.0 - albedo) / 4.0; }
int main(){
  std::cout << "scenario_name,model_type,equilibrium_temperature,adjustment_time,absorbed_solar,warning\n";
  std::cout << "baseline_one_layer,one_layer," << equilibrium_temperature(3.7,1.2) << "," << adjustment_time(10.0,1.2) << "," << absorbed_solar(1361.0,0.30) << ",boundaries_and_feedback_must_be_documented\n";
  std::cout << "stronger_feedback,one_layer," << equilibrium_temperature(3.7,1.8) << "," << adjustment_time(10.0,1.8) << "," << absorbed_solar(1361.0,0.30) << ",feedback_strength_changes_response\n";
}
